src/app/table/table.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit, Output, EventEmitter, ElementRef, HostListener } from '@angular/core';
+import { TableService } from '../services/table.service';
+
+
+@Component({
+  selector: 'app-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.css'],
+  providers: [TableService]
+})
+
+export class TableComponent implements OnInit {
+
+  paragraph:HTMLParagraphElement;
+  trElement:HTMLTableRowElement;
+  @Output('emitTrElement') sendDataEmitter = new EventEmitter<string>();
+  message: string;
+  users = this.tableService.users;
```

Comment:
to miejsce powinno służyć jedynie do deklaracji podstawowych rzeczy, odwołanie się do serwisu za pomocą `this` nie należy do nich więc powinniśmy wykonać je na OnInit -> kiedy zamienisz tableService aby używał `rxjs` wtedy zobaczysz większą wartość z tego rozwiązania